### ISSUE TITLE
Precendence of environment variables or configuration file values is causing unexpected issues

### ISSUE THREAD

**Issue by jwmatthews** [konveyor/kai#399]
As I am testing expected usage patterns of `podman compose up` I've run into 2 confusion points when a user supplies a `config.toml` to run with.

- #395 
- #397 

The issue in both of those cases is that for `podman compose up` we are specifying values in https://github.com/konveyor/kai/blob/main/compose.yaml which are intended to take highest precedence.  These values are transmitted via environment variables to the Kai process.

When we run with a specified configuration file, the path to the configuration file is being passed in as a command line argument in the entrypoint of the container:
https://github.com/konveyor/kai/blob/main/build/entrypoint.sh#L45

	if [[ -f /podman_compose/build/config.toml ]]; then
		printf "Using custom config file\n"
		PYTHONPATH="/kai/kai" python /kai/kai/server.py --config_filepath /podman_compose/build/config.toml
	else
		PYTHONPATH="/kai/kai" python /kai/kai/server.py
	fi

As this configuration file appears to be an explicit command line argument via the entrypoint we are interpreting the values in the configuration file to take higher precedence than what we have in environment variables.   

In the case of `podman compose up` this feels wrong, I think he User will assume that compose.yaml is the first source of truth.
I think they will assume the precedence is:
1. Environment variables
2. Configuration file

I don't think they will think that the configuration file they use ends up being a command line argument so it takes higher precedence. 

When I think of the user running `podman compose up`, my assumptions of their desires are: 
- `compose.yaml` takes precedence, this is where we set things like DB host, Demo Mode, Logging, etc
- `build/config.toml` is used to change models and/or perhaps algorithm choices, but primarily I see model choices as the main setting being changed

**Comment by shawn-hurley** [comment]
@Jonah do you think that #526 fixes this issue as well?

**Comment by JonahSussman** [comment]
Yes, it should. We're removing config.toml from the equation entirely.

**Comment by shawn-hurley** [comment]
closing because #526 is mergedff
